src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
     (ActorFunc)EnFr_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1A9A0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/EnFr_Init.s")
+static EnFrColor sEnFrColor[5] = { { 200, 170, 0, 255 },   // FROG_YELLOW
+                                   { 0, 170, 200, 255 },   // FROG_BLUE
+                                   { 210, 120, 100, 255 },   // FROG_RED
+                                   { 120, 130, 230, 255 },   // FROG_PURPLE
+                                   { 190, 190, 190, 255 } }; // FROG_WHITE
+
+// Jumping back into water frog animation
+//      sLogSpotToFromWater[frog].xzDist is magnitude of xz distance frog travels
+//      sLogSpotToFromWater[frog].yaw is rot around y-axis of jumping back into water
+//      sLogSpotToFromWater[frog].yDist is change in y distance frog travels
+static LogSpotToFromWater sLogSpotToFromWater[6] = {
+    { 0.0, 0.0, 0.0 },              // Prop   (Where link pulls ocarina)
+    { 80.0, -0.375 * M_PI, -80.0 }, // FROG_YELLOW
+    { 80.0, -0.5 * M_PI, -80.0 },   // FROG_BLUE
+    { 80.0, -0.25 * M_PI, -80.0 },  // FROG_RED
+    { 80.0, 0.875 * M_PI, -80.0 },  // FROG_PURPLE
+    { 80.0, 0.5 * M_PI, -80.0 }     // FROG_WHITE
+};
+
+// Timer values for the frog song
+static s16 sTimerFrogSong[] = { 40, // frames to play the first note
+                                20, // frames to play the second note
+                                15, // frames to play the third note
+                                12, // frames to play the fourth note
+                                12, // frames to play each remaining note
+                                0 };
+
+InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 2, ICHAIN_CONTINUE),
```

Comment:
make this static

Reply:
The reason I did not make this static was because it raised the error:
```
mips-linux-gnu-ld: build/src/overlays/actors/ovl_En_Fr/z_en_fr.o: in function `EnFr_Update':
(.text+0x2e0): undefined reference to `sInitChain'
mips-linux-gnu-ld: (.text+0x304): undefined reference to `sInitChain'
make: *** [Makefile:177: zelda_ocarina_mq_dbg.elf] Error 1
```

Reply:
oh, is update a non matching? if so then makes sense, its a limitation of our asm preprocessor. in which case i recommend reverting this back to the D_ name until the function it matched